### PR TITLE
Datanode List Page UI

### PR DESCRIPTION
**Pull request by gally47**
## Motivation and Context

fixes https://github.com/Graylog2/graylog2-server/issues/17159

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by grotlue — CHANGES_REQUESTED** [review]
Below are the results of testing this PR.
I still need to test the sorting functionality of the columns "Name" and "Is leader" and the searching but I still need to figure out how to manipulate these fields.
I found some issues with this list, I will review the code afterwards.

## Test results
✅Data Node List shows all data nodes with "Name", "Transport address", "Status", "Is leader", "Certificate valid until" and "Actions"
✅When clicking on "More" in the "Actions" column it shows the following actions
	- Renew certificate
	- Start
	- Stop
	- Rejoin
	- Remove
❌ The action "Stop" should only be shown for running data nodes
❌ The action "Start" should only be shown for stopped data nodes
✅When clicking on "Stop" it shows a notification
✅When clicking on "Stop" the status changes from "Available" to "Unavailable"
✅When clicking on "Start" it shows a notification
✅When clicking on "Start" the status changes from "Unavailable" to "Starting" to "Available"
❌ Entries can be sorted by "Status"
❌ Entries can be sorted by "	Certificate valid until"
❌ The search syntax help shows information about searchable fields in this list -> Shows fields that are not in the list, like `id`, `description`
✅Clicking on the name leads to the detail view
✅Columns can be hidden and shown using the "Columns" dropdown on the right
✅A "Data Nodes documentation" link on the top right redirects to the official Graylog Documentation for Data Nodes
✅When clicking "Renew certificate" it shows a success notification
When clicking "Renew certificate" the certificate is renewed
⚠️ The confirmation modal for "Rejoin" talks about resetting the Data Node. The naming is not consistent and thus confusing. I'd suggest to change "reset" to "rejoin" in the modal.
✅When clicking "Remove" a confirmation modal is shown
✅When confirming the removal a success notification is shown
✅When confirming the removal the status changes to "Removing" and later to "Removed"
✅When clicking "Rejoin" a confirmation modal is shown
✅When confirming rejoining a success notification is shown
✅When confirming rejoining the status changes to "Unconfigured"
❌ When confirming rejoining the status does not change further and stays "Unconfigured", however in preflight mode nothing else can be configured for this Data Node
❌ When removing a rejoined Data Node which is stuck on "Unconfigured" it shows a success message, but the status does not change 
✅Entries can be sorted by "Name"
✅Entries can be sorted by "Is leader"
❌ Entries can be searched by "Name"

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import fetch from 'logic/rest/FetchProvider';
+import type { Attribute, PaginatedListJSON, SearchParams } from 'stores/PaginationTypes';
+
+export const removeDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('DELETE', qualifyUrl(`/datanode/${datanodeId}`));
+
+    UserNotification.success(`Datanode "${datanodeId}" removed successfully`);
+  } catch (errorThrown) {
+    UserNotification.error(`Removing Datanode failed with status: ${errorThrown}`, 'Could not Remove the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:32]:
```suggestion
    UserNotification.error(`Removing Datanode failed with status: ${errorThrown}`, 'Could not remove the Datanode.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+export const stopDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('POST', qualifyUrl(`/datanode/${datanodeId}/stop`));
+
+    UserNotification.success(`Datanode "${datanodeId}" stopped successfully`);
+  } catch (errorThrown) {
+    UserNotification.error(`Stopping Datanode failed with status: ${errorThrown}`, 'Could not Stop the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:52]:
```suggestion
    UserNotification.error(`Stopping Datanode failed with status: ${errorThrown}`, 'Could not stop the Datanode.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+export const rejoinDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('POST', qualifyUrl(`/datanode/${datanodeId}/reset`));
+
+    UserNotification.success(`Datanode "${datanodeId}" rejoined successfully`);
+  } catch (errorThrown) {
+    UserNotification.error(`Rejoining Datanode failed with status: ${errorThrown}`, 'Could not Rejoin the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:62]:
```suggestion
    UserNotification.error(`Rejoining Datanode failed with status: ${errorThrown}`, 'Could not rejoin the Datanode.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  refetch: () => void,
+  isInitialLoading: boolean,
+} => {
+  const { data, refetch, isInitialLoading } = useQuery(
+    ['datanodes'],
+    () => fetchDataNodes(params),
+    {
+      onError: (errorThrown) => {
+        UserNotification.error(`Loading datanodes failed with status: ${errorThrown}`,
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:98]: Same here, another way to spell `datanodes`.

File: graylog2-web-interface/src/pages/DataNodesClusterPage.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+const DataNodesClusterPage = () => (
+  <DocumentTitle title="Data Nodes Cluster">
+    <DataNodesPageNavigation />
+    <PageHeader title="Data Nodes Cluster"
+                documentationLink={{
+                  title: 'Data Nodes documentation',
+                  path: DocsHelper.PAGES.GRAYLOG_DATA_NODE,
+                }}>
+      <span>
+        Graylog data nodes offer a better integration with Graylog and simplify future updates. They allow you to index and search through all the messages in your Graylog message database.
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesClusterPage.tsx:34]: Same with the spelling `data nodes`.

File: graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+const DataNodesMigrationPage = () => (
+  <DocumentTitle title="Data Nodes Migration">
+    <DataNodesPageNavigation />
+    <PageHeader title="Data Nodes Migration"
+                documentationLink={{
+                  title: 'Data Nodes documentation',
+                  path: DocsHelper.PAGES.GRAYLOG_DATA_NODE,
+                }}>
+      <span>
+        Graylog data nodes offer a better integration with Graylog and simplify future updates. They allow you to index and search through all the messages in your Graylog message database.
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx:33]: Same here. (spelling)

**Review by grotlue — CHANGES_REQUESTED** [review]
Code looks good in general. However, I think we should go through all the copy and make sure we use one spelling of `Data Node`.

**Comment by ousmaneo** [comment]
Thank you for your test and review. I fixed all the frontend-related comments. For these below, we will need to adjust the backend; for example, for the search-related issue, the frontend is sending the right request, same for the sorting.

> ❌ Entries can be sorted by "Status"
> ❌ Entries can be sorted by " Certificate valid until"
> ❌ When confirming rejoining the status does not change further and stays "Unconfigured", however in preflight mode nothing else can be configured for this Data Node
> ❌ When removing a rejoined Data Node which is stuck on "Unconfigured" it shows a success message, but the status does not change
> ❌ Entries can be searched by "Name"

**Comment by moesterheld** [comment]
> ❌ Entries can be sorted by "Status"
> ❌ Entries can be sorted by " Certificate valid until"

I have created a issue for these, as making that available is not trivial. Maybe the sorting button could be hidden in these columns in the UI in the meantime?

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+const DIALOG_TYPES = {
+  STOP: 'stop',
+  REJOIN: 'rejoin',
+  REMOVE: 'remove',
+  RENEW_CERT: 'renew',
+};
+
+const DIALOG_TEXT = {
+  [DIALOG_TYPES.REJOIN]: {
+    dialogTitle: 'Rejoin Datanode',
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:47]:
```suggestion
    dialogTitle: 'Rejoin Data Node',
```

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    dialogBody: (datanode: string) => `Are you sure you want to rejoin datanode "${datanode}"?`,
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:48]:
```suggestion
    dialogBody: (datanode: string) => `Are you sure you want to rejoin Data Node "${datanode}"?`,
```

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  },
+  [DIALOG_TYPES.REMOVE]: {
+    dialogTitle: 'Remove datanode',
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:51]:
```suggestion
    dialogTitle: 'Remove Data Node',
```

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    dialogBody: (datanode: string) => `Are you sure you want to remove datanode "${datanode}"?`,
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:52]:
```suggestion
    dialogBody: (datanode: string) => `Are you sure you want to remove Data Node "${datanode}"?`,
```

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+  },
+  [DIALOG_TYPES.STOP]: {
+    dialogTitle: 'Stop datanode',
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:55]:
```suggestion
    dialogTitle: 'Stop Data Node',
```

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+    dialogBody: (datanode: string) => `Are you sure you want to stop datanode "${datanode}"?`,
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:56]:
```suggestion
    dialogBody: (datanode: string) => `Are you sure you want to stop Data Node "${datanode}"?`,
```

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeList.tsx [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+                                     <p>
+                                       Find entities with a description containing node:<br />
+                                       <code>name:node</code><br />
+                                     </p>
+                                   )} />
+                    )} />
+      </SearchContainer>
+      <div>
+        {elements?.length === 0 ? (
+          <NoSearchResult>No Data Nodes have been found</NoSearchResult>
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeList.tsx:143]: Here the spelling would already fit 👍

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+import type { DataNode } from 'preflight/types';
+import UserNotification from 'util/UserNotification';
+import fetch from 'logic/rest/FetchProvider';
+import type { Attribute, PaginatedListJSON, SearchParams } from 'stores/PaginationTypes';
+
+export const removeDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('DELETE', qualifyUrl(`/datanode/${datanodeId}`));
+
+    UserNotification.success(`Datanode "${datanodeId}" removed successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:30]:
```suggestion
    UserNotification.success(`Data Node "${datanodeId}" removed successfully`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+  } catch (errorThrown) {
+    UserNotification.error(`Removing Data node failed with status: ${errorThrown}`, 'Could not remove the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:32]:
```suggestion
    UserNotification.error(`Removing Data Node failed with status: ${errorThrown}`, 'Could not remove the Data Node.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+export const startDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('POST', qualifyUrl(`/datanode/${datanodeId}/start`));
+
+    UserNotification.success(`Data node "${datanodeId}" started successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:40]:
```suggestion
    UserNotification.success(`Data Node "${datanodeId}" started successfully`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+  } catch (errorThrown) {
+    UserNotification.error(`Starting Data node failed with status: ${errorThrown}`, 'Could not start the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:42]:
```suggestion
    UserNotification.error(`Starting Data Node failed with status: ${errorThrown}`, 'Could not start the Data Node.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+export const stopDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('POST', qualifyUrl(`/datanode/${datanodeId}/stop`));
+
+    UserNotification.success(`Data node "${datanodeId}" stopped successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:50]:
```suggestion
    UserNotification.success(`Data Node "${datanodeId}" stopped successfully`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+  } catch (errorThrown) {
+    UserNotification.error(`Stopping Data node failed with status: ${errorThrown}`, 'Could not stop the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:52]:
```suggestion
    UserNotification.error(`Stopping Data Node failed with status: ${errorThrown}`, 'Could not stop the Data Node.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+export const rejoinDataNode = async (datanodeId: string) => {
+  try {
+    await fetch('POST', qualifyUrl(`/datanode/${datanodeId}/reset`));
+
+    UserNotification.success(`Data node "${datanodeId}" rejoined successfully`);
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:60]:
```suggestion
    UserNotification.success(`Data Node "${datanodeId}" rejoined successfully`);
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 21]
```diff
@@ -0,0 +1,118 @@
...
+  } catch (errorThrown) {
+    UserNotification.error(`Rejoining Data node failed with status: ${errorThrown}`, 'Could not rejoin the Datanode.');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:62]:
```suggestion
    UserNotification.error(`Rejoining Data Node failed with status: ${errorThrown}`, 'Could not rejoin the Data Node.');
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  refetch: () => void,
+  isInitialLoading: boolean,
+} => {
+  const { data, refetch, isInitialLoading } = useQuery(
+    ['datanodes'],
+    () => fetchDataNodes(params),
+    {
+      onError: (errorThrown) => {
+        UserNotification.error(`Loading data nodes failed with status: ${errorThrown}`,
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:98]:
```suggestion
        UserNotification.error(`Loading Data Nodes failed with status: ${errorThrown}`,
```

File: graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts [hunk 23]
```diff
@@ -0,0 +1,118 @@
...
+          'Could not load data nodes');
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/hooks/useDataNodes.ts:99]:
```suggestion
          'Could not load Data Nodes');
```

File: graylog2-web-interface/src/pages/DataNodesClusterPage.tsx [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+const DataNodesClusterPage = () => (
+  <DocumentTitle title="Data nodes Cluster">
+    <DataNodesPageNavigation />
+    <PageHeader title="Data nodes Cluster"
+                documentationLink={{
+                  title: 'Data nodes documentation',
+                  path: DocsHelper.PAGES.GRAYLOG_DATA_NODE,
+                }}>
+      <span>
+        Graylog data nodes offer a better integration with Graylog and simplify future updates. They allow you to index and search through all the messages in your Graylog message database.
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesClusterPage.tsx:34]:
```suggestion
        Graylog Data Nodes offer a better integration with Graylog and simplify future updates. They allow you to index and search through all the messages in your Graylog message database.
```

File: graylog2-web-interface/src/pages/DataNodesClusterPage.tsx [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+import DocsHelper from 'util/DocsHelper';
+import DataNodeConfiguration from 'components/datanode/DataNodeConfiguration/DataNodeConfiguration';
+import DataNodesPageNavigation from 'components/datanode/DataNodePageNavigation';
+
+const DataNodesClusterPage = () => (
+  <DocumentTitle title="Data nodes Cluster">
+    <DataNodesPageNavigation />
+    <PageHeader title="Data nodes Cluster"
+                documentationLink={{
+                  title: 'Data nodes documentation',
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesClusterPage.tsx:30]:
```suggestion
                  title: 'Data Nodes documentation',
```

File: graylog2-web-interface/src/pages/DataNodesClusterPage.tsx [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+import React from 'react';
+
+import { Row, Col } from 'components/bootstrap';
+import { DocumentTitle, PageHeader } from 'components/common';
+import DocsHelper from 'util/DocsHelper';
+import DataNodeConfiguration from 'components/datanode/DataNodeConfiguration/DataNodeConfiguration';
+import DataNodesPageNavigation from 'components/datanode/DataNodePageNavigation';
+
+const DataNodesClusterPage = () => (
+  <DocumentTitle title="Data nodes Cluster">
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesClusterPage.tsx:26]:
```suggestion
  <DocumentTitle title="Data Nodes Cluster">
```

File: graylog2-web-interface/src/pages/DataNodesClusterPage.tsx [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+    <DataNodesPageNavigation />
+    <PageHeader title="Data nodes Cluster"
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesClusterPage.tsx:28]:
```suggestion
    <PageHeader title="Data Nodes Cluster"
```

File: graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+ */
+import React from 'react';
+
+import { Row, Col } from 'components/bootstrap';
+import { DocumentTitle, PageHeader } from 'components/common';
+import DocsHelper from 'util/DocsHelper';
+import DataNodesPageNavigation from 'components/datanode/DataNodePageNavigation';
+
+const DataNodesMigrationPage = () => (
+  <DocumentTitle title="Data nodes Migration">
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx:25]:
```suggestion
  <DocumentTitle title="Data Nodes Migration">
```

File: graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+    <DataNodesPageNavigation />
+    <PageHeader title="Data nodes Migration"
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx:27]:
```suggestion
    <PageHeader title="Data Nodes Migration"
```

File: graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+                documentationLink={{
+                  title: 'Data nodes documentation',
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx:29]:
```suggestion
                  title: 'Data Nodes documentation',
```

File: graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+                  path: DocsHelper.PAGES.GRAYLOG_DATA_NODE,
+                }}>
+      <span>
+        Graylog data nodes offer a better integration with Graylog and simplify future updates. They allow you to index and search through all the messages in your Graylog message database.
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesMigrationPage.tsx:33]:
```suggestion
        Graylog Data Nodes offer a better integration with Graylog and simplify future updates. They allow you to index and search through all the messages in your Graylog message database.
```

File: graylog2-web-interface/src/pages/DataNodesPage.tsx [hunk 32]
```diff
@@ -19,10 +19,12 @@ import React from 'react';
...
-import DataNodeConfiguration from 'components/datanode/DataNodeConfiguration/DataNodeConfiguration';
+import DataNodesPageNavigation from 'components/datanode/DataNodePageNavigation';
+import DataNodeList from 'components/datanode/DataNodeList/DataNodeList';
 
 const DataNodesPage = () => (
   <DocumentTitle title="Data Nodes">
+    <DataNodesPageNavigation />
     <PageHeader title="Data Nodes"
                 documentationLink={{
                   title: 'Data Nodes documentation',
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesPage.tsx:30]: Spelling already correct.

File: graylog2-web-interface/src/pages/DataNodesPage.tsx [hunk 33]
```diff
@@ -19,10 +19,12 @@ import React from 'react';
...
 import { DocumentTitle, PageHeader } from 'components/common';
 import DocsHelper from 'util/DocsHelper';
-import DataNodeConfiguration from 'components/datanode/DataNodeConfiguration/DataNodeConfiguration';
+import DataNodesPageNavigation from 'components/datanode/DataNodePageNavigation';
+import DataNodeList from 'components/datanode/DataNodeList/DataNodeList';
 
 const DataNodesPage = () => (
   <DocumentTitle title="Data Nodes">
+    <DataNodesPageNavigation />
     <PageHeader title="Data Nodes"
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/DataNodesPage.tsx:28]: Spelling already correct.

File: graylog2-web-interface/src/components/datanode/DataNodePageNavigation.tsx [hunk 34]
```diff
@@ -21,9 +21,9 @@ import Routes from 'routing/Routes';
 import { Row } from 'components/bootstrap';
 
 const NAV_ITEMS = [
-  { title: 'Overview', path: Routes.SYSTEM.DATANODES.OVERVIEW, exactPathMatch: true },
-  { title: 'Monitoring & Management', path: Routes.SYSTEM.DATANODES.OVERVIEW },
-  { title: 'Migrations', path: Routes.SYSTEM.DATANODES.OVERVIEW },
+  { title: 'Data Nodes', path: Routes.SYSTEM.DATANODES.LIST, exactPathMatch: true },
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodePageNavigation.tsx:24]: Spelling already correct.

**Review by grotlue — CHANGES_REQUESTED** [review]
Thanks a lot for the fixes!

I still found inconsistencies in the copy, so I went through all the files and commented suggestions so it's easier to track an update. I suggested to go with `Data Node`/`Data Nodes`, using it as a term/own name.


> > ❌ Entries can be sorted by "Status"
> > ❌ Entries can be sorted by " Certificate valid until"
> 
> I have created a issue for these, as making that available is not trivial. Maybe the sorting button could be hidden in these columns in the UI in the meantime?

I think this is a good idea, so we don't merge something broken.

I retested the other cases and they look fine now! However, I added to more cases which I think should be adjusted.

### Test results
✅ Entries can be searched by "Name"
✅ When confirming rejoining the status does not change further and stays "Unconfigured", however in preflight mode nothing else can be configured for this Data Node
✅ ~When removing a rejoined Data Node which is stuck on "Unconfigured" it shows a success message, but the status does not change~
✅ The search syntax help shows information about searchable fields in this list -> Shows fields that are not in the list, like id, description
✅ The action "Stop" should only be shown for running data nodes
✅ The action "Start" should only be shown for stopped data nodes
✅ ~The confirmation modal for "Rejoin" talks about resetting the Data Node. The naming is not consistent and thus confusing. I'd suggest to change "reset" to "rejoin" in the modal.~

❌ The action "Rejoin" should only be shown for removed data nodes
❌ The action "Remove" should only be for data nodes that are not already removed

File: graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx [hunk 35]
```diff
@@ -0,0 +1,144 @@
...
+const DIALOG_TYPES = {
+  STOP: 'stop',
+  REJOIN: 'rejoin',
+  REMOVE: 'remove',
+  RENEW_CERT: 'renew',
+};
+
+const DIALOG_TEXT = {
+  [DIALOG_TYPES.REJOIN]: {
+    dialogTitle: 'Rejoin Data node',
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx:47]:
```suggestion
    dialogTitle: 'Rejoin Data Node',
```

**Review by grotlue — COMMENTED** [review]
Two more places for the spelling. I will just commit those suggestions, that will make it easier :)

Also It would be good if we can still "fix":
❌ The action "Rejoin" should only be shown for removed data nodes
❌ The action "Remove" should only be for data nodes that are not already removed

**Review by grotlue — APPROVED** [review]
Looking great! Thanks a lot!

✅ The action “Rejoin” should only be shown for removed data nodes
✅ The action “Remove” should only be for data nodes that are not already removed

**Pull request merged**